tracing-opentelemetry/src/layer.rs

```diff
@@ -153,11 +155,37 @@ impl<'a> field::Visit for SpanAttributeVisitor<'a> {
     ///
     /// [`Span`]: https://docs.rs/opentelemetry/latest/opentelemetry/api/trace/span/trait.Span.html
     fn record_debug(&mut self, field: &field::Field, value: &dyn fmt::Debug) {
-        let attribute = api::Key::new(field.name()).string(format!("{:?}", value));
-        if let Some(attributes) = &mut self.0.attributes {
-            attributes.push(attribute);
+        if field.name() == SPAN_NAME_FIELD {
+            self.0.name = format!("{:?}", value);
         } else {
```

Comment:
It looks like `SpanNameVisitor` handles this case in both `record_debug` _and_ `record_str`, while here, we only handle it in `record_debug`? That means that if a string value is used as the name in a call to `Span::record` after the span is created, we will format it using `fmt::Debug`. The `Debug` formatter for strings includes quotes and escapes special characters, which may not be the right behavior for the span name?

Reply:
added a `record_str` impl 👍